core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -122,7 +125,26 @@ object JmxTool extends Logging {
       else
         List(null)
 
-    val names = queries.flatMap((name: ObjectName) => mbsc.queryNames(name, null).asScala)
+    val hasPatternQueries = queries.exists((name: ObjectName) => name.isPattern)
+
+    retries = 0
+    var names : Iterable[ObjectName] = null
+    if (!hasPatternQueries) {
+      do {
+        if (names != null) {
+          System.err.println("Could not find all object names, retrying")
+          retries += 1
+          Thread.sleep(500)
+        }
+        names = queries.flatMap((name: ObjectName) => mbsc.queryNames(name, null).asScala)
+      } while (wait && retries < maxNumRetries && !queries.equals(names))
```

Comment:
The `queries.equals(names)` check is a bit fragile as the collection type is `Iterable`. If one of them is a `Set` and the other is a `Seq`, they will always be false, for example. Maybe we should make sure both instances are a `Set` before comparing. It also seems to make more sense since we don't care about ordering.